### PR TITLE
CDPCP-2598 Add logs around major user sync tasks

### PR DESCRIPTION
**Pull request by handavid**
Add logs around the major user sync tasks so that we can
easily determine duration of these tasks from the logs.

See detailed description in the commit message.

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java [hunk 1]
```diff
@@ -208,8 +208,10 @@ private void internalSynchronizeUsers(String operationId, String accountId, Stri
                     umsEventGenerationIdsProvider.getEventGenerationIds(accountId, requestId) :
                     null;
 
+            LOGGER.debug("Starting RETRIEVE_{}_UMS_STATE for environments {} ...", fullSync ? "FULL" : "PARTIAL", environmentCrns);
```
Comment (giladwolff) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java:211]: Is there a request ID in this context? If there is, is it included in the message? Maybe add the operation ID?

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java [hunk 2]
```diff
@@ -208,8 +208,10 @@ private void internalSynchronizeUsers(String operationId, String accountId, Stri
...
             Map<String, UmsUsersState> envToUmsStateMap = umsUsersStateProvider
                     .getEnvToUmsUsersStateMap(accountId, actorCrn, environmentCrns, userCrnFilter, machineUserCrnFilter, requestId);
+            LOGGER.debug("Finished RETRIEVE_{}_UMS_STATE.", fullSync ? "FULL" : "PARTIAL");
```
Comment (giladwolff) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java:214]: and here?

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java [hunk 3]
```diff
@@ -276,18 +278,26 @@ private SyncStatusDetail internalSynchronizeStack(Stack stack, UmsUsersState ums
         Multimap<String, String> warnings = ArrayListMultimap.create();
         try {
             FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStack(stack);
+            LOGGER.debug("Starting RETRIEVE_{}_IPA_STATE ...", fullSync ? "FULL" : "PARTIAL");
             UsersState ipaUsersState = getIpaUserState(freeIpaClient, umsUsersState, fullSync);
-            LOGGER.debug("IPA UsersState, found {} users and {} groups", ipaUsersState.getUsers().size(), ipaUsersState.getGroups().size());
+            LOGGER.debug("Finished RETRIEVE_{}_IPA_STATE, found {} users and {} groups.", fullSync ? "FULL" : "PARTIAL",
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java:283]: this `FULL` vs `PARTIAL` is duplicated. maybe it should be move to a variable or a method

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java (hunk 1 above)
Comment (handavid) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java:211]: the request id and operation id (exposed as "flow") are part of the MDC.
```
2020-07-13 14:42:58,291 [usersyncExecutor-1] lambda$internalSynchronizeUsers$7:204 DEBUG c.s.f.s.f.u.UserSyncService - [instance:default] [type:springLog] [crn:] [name:] [flow:ab77043f-84e5-49d1-9e57-fa27e8393a9f] [requestid:2bc3eed0-77ff-4ab2-b5a2-2ce4fd17da4c] [tenant:] [userCrn:] [environment:] [traceId:] [spanId:] Starting RETRIEVE_FULL_UMS_STATE for environments [crn:cdp:environments:us-west-1:foo:environment:8813ff3f-b25b-40f8-9999-569804d48396] ...
```

**Pull request merged**